samples/EverythingServer/LoggingUpdateMessageSender.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ModelContextProtocol.Protocol.Messages;
+using ModelContextProtocol.Protocol.Types;
+using ModelContextProtocol.Server;
+
+namespace EverythingServer;
+
+public class LoggingUpdateMessageSender(IMcpServer server) : IHostedService
+{
+    readonly Dictionary<LoggingLevel, string> _loggingLevelMap = new()
+    {
+        { LoggingLevel.Debug, "Debug-level message" },
+        { LoggingLevel.Info, "Info-level message" },
+        { LoggingLevel.Notice, "Notice-level message" },
+        { LoggingLevel.Warning, "Warning-level message" },
+        { LoggingLevel.Error, "Error-level message" },
+        { LoggingLevel.Critical, "Critical-level message" },
+        { LoggingLevel.Alert, "Alert-level message" },
+        { LoggingLevel.Emergency, "Emergency-level message" }
+    };
+
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        var currentLevel = server.Services!.GetRequiredService<Func<LoggingLevel>>();
```

Comment:
While I'm generally not a fan of registering `Func`'s directly with DI, I can appreciate the convenience for a sample. We should use constructor injection though. For example:

```csharp
public class LoggingUpdateMessageSender(IMcpServer server, Func<LoggingLevel> getLogLevel) : BackgroundService
```

Reply:
I couldn't think of a better way than using a `Func` to access the mutable value. An alternative would be to have it resolve on every loop iteration using `server.Services`, would that be preferrable?

Reply:
I'm okay with using the `Func` since it's sample code. We should still make it a constructor parameter instead of calling GetRequiredService.